### PR TITLE
fix: trapped_signal: unbound variable.

### PR DESCRIPTION
**Pull request by CanmingCobble**
## Description

Recursion reports the following problem.  This defect is particularly about shell error in signal handling.

[2023-01-25 22:17:07] || INFO: [152214] root: Validating checkpoint storage ...
<info> [2023-01-25 22:17:10] || INFO: [152214] root: Uploading to GCS: determined/f778d383-3577-4220-915c-ec34a7fd7625
<error> [2023-01-25 22:17:15] || ERROR: [152214] root: Checkpoint storage validation failed: ('invalid_grant: Bad Request', {'error': 'invalid_grant', 'error_description': 'Bad Request'})
<none> [2023-01-25 22:17:15] || /run/determined/task-signal-handling.sh: line 39: trapped_signal: unbound variable

Apparently, recursion introduced a 'set -u' in their startup-hook.sh to trigger this.   That impacts the method invoked by the trap handler when a trap occurs apparently.

## Test Plan
A reproducer:

```
#!/bin/bash

set -u

myvar="yes"
if [ -z "${myvar}" ] ; then
   echo "1 - myvar Not defined"
else
   echo "1 - myvar Defined"
fi

unset myvar
if [ -z "${myvar+x}" ] ; then
   echo "2 - myvar Not defined"
else
   echo "2 - myvar Defined"
fi

if [ -z "${myvar}" ] ; then
   echo "3 - myvar Not defined"
else
   echo "3 - myvar Defined"
fi

```

% bash test.sh
1 - myvar Defined
2 - myvar Not defined
test.sh: line 19: myvar: unbound variable

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: master/static/srv/task-signal-handling.sh [hunk 1]
```diff
@@ -36,7 +36,7 @@ wait_and_handle_signals() {
         # When a signal is sent to the shell, it will interrupt waits, after all traps have run. To
         # discern if the wait unblocked because of a signal or process exit, we set "trapped_signal"
         # in traps and check it here.
-        if [ -z "${trapped_signal}" ]; then
+        if [ -z ${var+trapped_signal} ]; then
```
Comment (jerryharrow) [inline master/static/srv/task-signal-handling.sh:39]: This works (I manually tested it), but oddly doesn't seem to be documented for bash - https://www.gnu.org/software/bash/manual/bash.html#Shell-Parameter-Expansion

It is documented in the opengroup shell specification https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#tag_18_06_02

**Review by jerryharrow — COMMENTED** [review]
You should include in the PR description that this can apparently be triggered by a startup-hook.sh which does a `set -u`.

There is another variable reference on line 8 that likewise should probably be protected.

**Review by jerryharrow — APPROVED** [review]

**Comment by CanmingCobble** [comment]
Here is another case of unbound variable, after I add "set -u" to task-signal-handling.sh:
/run/determined/task-logging-setup.sh: line 27: DET_K8S_LOG_TO_FILE: unbound variable

**Comment by CanmingCobble** [comment]
> Here is another case of unbound variable, after I add "set -u" to task-signal-handling.sh: /run/determined/task-logging-setup.sh: line 27: DET_K8S_LOG_TO_FILE: unbound variable

Talked to Jerry, adding "set -u" to DAI code is not desired and we limit the scope to what customers can cause it to set for us. Which is via startup-hook.sh.

**Pull request merged**